src/main/java/com/sap/oss/phosphor/fosstars/data/release/info/ReleaseInfoLoader.java

```diff
@@ -0,0 +1,121 @@
+package com.sap.oss.phosphor.fosstars.data.release.info;
+
+import com.sap.oss.phosphor.fosstars.data.AbstractReleaseInfoLoader;
+import com.sap.oss.phosphor.fosstars.data.DataProvider;
+import com.sap.oss.phosphor.fosstars.data.github.GitHubDataFetcher;
+import com.sap.oss.phosphor.fosstars.data.github.ReleasesFromGitHub;
+import com.sap.oss.phosphor.fosstars.model.Value;
+import com.sap.oss.phosphor.fosstars.model.ValueSet;
+import com.sap.oss.phosphor.fosstars.model.subject.oss.Artifact;
+import com.sap.oss.phosphor.fosstars.model.subject.oss.GitHubProject;
+import com.sap.oss.phosphor.fosstars.model.subject.oss.MavenArtifact;
+import com.sap.oss.phosphor.fosstars.model.subject.oss.NpmArtifact;
+import com.sap.oss.phosphor.fosstars.model.value.ArtifactVersions;
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * This data provider tries to fill out the
+ * {@link com.sap.oss.phosphor.fosstars.model.feature.oss.OssFeatures#RELEASED_ARTIFACT_VERSIONS}
+ * feature. It is based on one of the following data providers:
+ * <ul>
+ * <li>{@link ReleaseInfoFromMaven}</li>
+ * <li>{@link ReleaseInfoFromNpm}</li>
+ * </ul>
+ * The data provider uses {@link ReleasesFromGitHub} as a fallback if
+ * {@link com.sap.oss.phosphor.fosstars.model.feature.oss.OssFeatures#RELEASED_ARTIFACT_VERSIONS}
+ * feature could not be filled up from one of the above-mentioned data providers.
+ */
+public class ReleaseInfoLoader extends AbstractReleaseInfoLoader<Artifact> {
+
+  /**
+   * A {@link ReleasesFromGitHub} data provider.
+   */
+  private final DataProvider<GitHubProject> releasesFromGitHub;
+  
+  /**
+   * A {@link ReleaseInfoFromMaven} data provider.
+   */
+  private final DataProvider<MavenArtifact> releaseInfoFromMaven;
+  
+  /**
+   * A {@link ReleaseInfoFromNpm} data provider.
+   */
+  private final DataProvider<NpmArtifact> releaseInfoFromNpm;
+
+  /**
+   * Initializes a data provider.
+   *
+   * @param fetcher An interface to GitHub.
+   */
+  public ReleaseInfoLoader(GitHubDataFetcher fetcher) {
+    Objects.requireNonNull(fetcher, "Oh no! You gave me a null instead of a GitHub fetcher!");
+    releasesFromGitHub = new ReleasesFromGitHub(fetcher);
+    releaseInfoFromMaven = new ReleaseInfoFromMaven();
+    releaseInfoFromNpm = new ReleaseInfoFromNpm();
+  }
+
+  @Override
+  public DataProvider<Artifact> update(Artifact artifact, ValueSet values) throws IOException {
+    doUpdate(artifact, values);
+    for (Value<?> value : values) {
+      if (value.get() instanceof ArtifactVersions && !((ArtifactVersions) value.get()).empty()) {
```

Comment:
What if there is another feature that holds `ArtifactVersions`? I am not sure that this method will work correctly in this case.

I don't think this loop is really necessary here. I believe you just need a single if-else block like in `doUpdate(Artifact artifact, ValueSet values)`. If necessary, you can mock the underlying data providers in the tests (you may want to add a constructor that takes sub-providers). Both `doUpdate` methods look redundant -- they can be just merged to `update()`.